moduleframework/tools/modulelint.py

```diff
@@ -43,57 +43,75 @@ def setUp(self):
         if self.dp.dockerfile is None:
             self.skip()
 
-    def testArchitectureInEnvAndLabelExists(self):
+    def test_architecture_in_env_and_label_exists(self):
         self.assertTrue(self.dp.get_docker_specific_env("ARCH="))
         self.assertTrue(self.dp.get_specific_label("architecture"))
 
-    def testNameInEnvAndLabelExists(self):
+    def test_name_in_env_and_label_exists(self):
         self.assertTrue(self.dp.get_docker_specific_env("NAME="))
         self.assertTrue(self.dp.get_specific_label("name"))
 
-    def testReleaseLabelExists(self):
+    def test_release_label_xists(self):
         self.assertTrue(self.dp.get_specific_label("release"))
 
-    def testVersionLabelExists(self):
+    def test_version_label_exists(self):
         self.assertTrue(self.dp.get_specific_label("version"))
 
-    def testComRedHatComponentLabelExists(self):
+    def test_com_redhat_component_label_exists(self):
         self.assertTrue(self.dp.get_specific_label("com.redhat.component"))
 
-    def testIok8sDescriptionExists(self):
-        self.assertTrue(self.dp.get_specific_label("io.k8s.description"))
 
-    def testIoOpenshiftExposeServicesExists(self):
-        label_io_openshift = "io.openshift.expose-services"
-        exposes = self.dp.get_docker_expose()
-        label_list = self.dp.get_docker_labels()
-        self.assertTrue(label_list[label_io_openshift])
-        for exp in exposes:
-            self.assertTrue("%s" % exp in label_list[label_io_openshift])
+class DockerfileLinterInContainer(container_avocado_test.ContainerAvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    def test_docker_nodocs(self):
+        self.start()
+        installed_pkgs = self.run("rpm -qa --qf '%{{NAME}}\n'", verbose=False).stdout
+        # This returns a list of packages defined in config.yaml for testing
+        # e.g. ["bash", "rpm", "memcached"] in case of memcached
+        pkgs = self.backend.getPackageList()
+        list_pkg = [pkg for pkg in installed_pkgs.split('\n') if pkg in pkgs]
+        for pkg in list_pkg:
+            all_docs = self.run("rpm -qd %s" % pkg, verbose=False).stdout
+            for doc in all_docs.strip().split('\n'):
+                self.assertNotEqual(0, self.run("test -e %s" % doc, ignore_status=True).exit_status)
+
+    def test_docker_clean_all(self):
+        self.start()
+        pkg_mgr = "yum"
+        # Detect distro in image
+        distro = self.run("cat /etc/os-release").stdout
 
-    def testIoOpenShiftTagsExists(self):
-        label_list = self.dp.get_docker_labels()
-        self.assertTrue("io.openshift.tags" in label_list)
+        if 'NAME=Fedora' in distro:
+            pkg_mgr = "dnf"
+        # Look, whether we have solv files in /var/cache/<pkg_mgr>/*.solv
+        # dnf|yum clean all deletes the file *.solv
+        ret = self.run("ls /var/cache/%s/*.solv" % pkg_mgr, ignore_status=True)
```

Comment:
as I said previously, this check is totally wrong. yum doesn't have solv files for example. also this ties tests to fedora.

Reply:
This is fixed by some deterministic size, like proposing @TomasTomecek